src/NodeTypeResolver/TypeAnalyzer/ArrayTypeAnalyzer.php

```diff
@@ -5,104 +5,18 @@
 namespace Rector\NodeTypeResolver\TypeAnalyzer;
 
 use PhpParser\Node\Expr;
-use PhpParser\Node\Expr\Array_;
-use PhpParser\Node\Expr\PropertyFetch;
-use PhpParser\Node\Expr\StaticPropertyFetch;
-use PHPStan\Reflection\ClassReflection;
-use PHPStan\Reflection\Php\PhpPropertyReflection;
-use PHPStan\Type\Accessory\HasOffsetType;
-use PHPStan\Type\Accessory\NonEmptyArrayType;
-use PHPStan\Type\ArrayType;
-use PHPStan\Type\IntersectionType;
-use PHPStan\Type\MixedType;
-use PHPStan\Type\Type;
 use Rector\NodeTypeResolver\NodeTypeResolver;
-use Rector\Reflection\ReflectionResolver;
 
 final readonly class ArrayTypeAnalyzer
```

Comment:
Maybe this whole class may now be deprecated?

Reply:
I think we can keep it for now in case of regression, then we can add some logic here instead